src/site/content/en/blog/receive-shared-data/index.md

```diff
@@ -0,0 +1,262 @@
+    }
+  }
+}
+```
+
+### Accepting files
+
+As with application changes, accepting files requires that `method` be `"POST"`
+and that `enctype` be present. Additionally, `enctype` must be
+`"multipart/form-data"`, and a `files` entry must be added.
+
+You must also add a `files` array defining the types of files your app accepts. The
+array elements are entries with two members: a `name` field and an `accept`
+field. The `accept` field takes a MIME type, a file extension, or an array
+containing both.
+
+```json/5,10-19
+{
+  "name": "Aggregator",
+  "share_target": {
+    "action": "/cgi-bin/aggregate",
+    "method": "POST",
+    "enctype": "multipart/form-data",
+    "params": {
+      "title": "name",
+      "text": "description",
+      "url": "link",
+      "files": [
+        {
+          "name": "records",
+          "accept": ["text/csv", ".csv"]
+        },
+        {
+          "name": "graphs",
+          "accept": "image/svg+xml"
+        }
+      ]
+    }
+  }
+}
+```
+
+## Handle the incoming content
+
+How you deal with the incoming shared data is up to you and depends on your
+app. For example:
+
+* An email client could draft a new email using `title` as the subject of an
+  email, with `text` and `url` concatenated together as the body.
+* A social networking app could draft a new post ignoring `title`, using
+  `text` as the body of the message, and adding `url` as a link. If `text` is
+  missing, the app might use `url` in the body as well. If `url` is missing,
+  the app might scan `text` looking for a URL and add that as a link.
+* A photo sharing app could create a new slideshow using `title` as the
+  slideshow title, `text` as a description, and `files` as the slideshow images.
+* A text messaging app could draft a new message using `text` and `url`
+  concatenated together and dropping `title`.
+
+### Processing GET shares
+
+If the user selects your application, and your `method` is `"GET"` (the
+default), the browser opens a new window at the `action` URL. The browser then
+generates a query string using the URL-encoded values supplied in the manifest.
+For example, if the sharing app provides `title` and `text`, the query string is
+`?title=hello&text=world`. To process this, use a `DOMContentLoaded` event
+listener in your foreground page and parse the query string:
+
+```js
+window.addEventListener('DOMContentLoaded', () => {
+  const parsedUrl = new URL(window.location);
+  // searchParams.get() will properly handle decoding the values.
+  console.log('Title shared: ' + parsedUrl.searchParams.get('title'));
+  console.log('Text shared: ' + parsedUrl.searchParams.get('text'));
+  console.log('URL shared: ' + parsedUrl.searchParams.get('url'));
+});
+```
+
+Be sure to use a service worker to precache the `action` page so that it will
+load quickly and work reliably, even if the user is offline.
+
+### Processing POST shares
+
+If your `method` is `"POST"`, as it would be if it accepted a saved bookmark or
```

Comment:
I'm not sure what the "it" is in "if it accepted."

Reply:
Fixed.